src/core/utils/net.ts

```diff
@@ -2,24 +2,51 @@ import chalk from "chalk";
 import net from "net";
 import ora from "ora";
 import waitOn from "wait-on";
+import getPort from "get-port";
+import { DEFAULT_CONFIG } from "../../config";
+import { confirmChooseRandomPort } from "../prompts";
 import { logger } from "./logger";
 
+const VALID_PORT_MIN = 1024;
+const VALID_PORT_MAX = 65535;
+
 /**
  * Check if a given remote address and port are accepting TCP connections.
  * @param Object host and port of the server to check.
- * @returns True if the given server is accepting TCP connections. False otherwise.
+ * @returns The resolved port number if the given server is accepting TCP connections. 0 if the port is already taken.
  */
-export function isAcceptingTcpConnections({ host = "127.0.0.1", port = 80 }: { host?: string; port?: number }) {
-  return new Promise<boolean>((resolve) => {
-    const socket = net.createConnection(port, host);
+export function isAcceptingTcpConnections({
+  host = DEFAULT_CONFIG.host,
+  port = DEFAULT_CONFIG.port,
+}: {
+  host?: string;
+  port?: number;
+}): Promise<number> {
+  port = Number(port) as number;
+  logger.silly(`Checking if ${host}:${port} is accepting TCP connections...`);
+
+  // Check if current port is beyonf the MAX valid port range
+  if (port > VALID_PORT_MAX) {
+    logger.silly(`Port ${port} is beyond the valid port range (${VALID_PORT_MAX}).`);
+    return Promise.resolve(0);
+  }
+
+  return new Promise((resolve) => {
+    const socket = net.createConnection(port as number, host);
 
     socket
       .once("error", () => {
-        resolve(false);
         socket.end();
+        logger.silly(`Port ${port} is available. Use it.`);
+        resolve(port as number);
       })
-      .once("connect", () => {
-        resolve(true);
+      .once("connect", async () => {
+        logger.warn(`Port ${port} is already taken!`);
+        const portNumber = await confirmChooseRandomPort();
+
+        if (portNumber) {
+          resolve(await getPort());
```

Comment:
No need for await, you can resolve a promise directly
```suggestion
          resolve(getPort());
```